---
title: "bartools: scRNA-seq analysis guide"
author: "Henrietta Holze"
output: 
  rmarkdown::html_document:
    toc: true
    number_sections: true
    toc_depth: 3
    highlight: tango 
date: September 05, 2023
vignette: >
  %\VignetteIndexEntry{bartools-single-cell-analysis}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: sentence
---

# Bartools single-cell guide

The `bartools` package contains methods to simplify clone level analyses from single-cell cellular barcoding datasets.
The purpose of this vignette is to highlight these capabilities of `bartools`. 

## 0. Setup
This vignette makes use of a simple test single cell experiment object (SCE) included in the `bartools` package. The dataset contains 100 cells each with 100 randomly sampled genes. Each cell has lineage barcode information annotated in the `test.sce$barcode` field. 

Dimensionality reduction and clustering were previously performed using `Seurat`. Cluster assignments are in the `test.sce$seurat_clusters` field.

### Load the `bartools` package

```{r}
library(bartools)
knitr::opts_chunk$set(dev="png")
```

### Load the `test.sce` dataset

```{r}
data(test.sce)
test.sce
```


## 1. Analyse dataset metrics using plotMetrics

Single cell RNA sequencing datasets can reveal transcriptional differences between groups of cells / celltypes that are of biological importance. The `plotMetrics` function takes a single cell object in Seurat or SingleCellExperiment format and plots a desired continuous variable (e.g. number of transcripts or genes detected per cell) split across any groups of interest. These groups could be clusters, celltypes or, with lineage barcode information, individual clones. Thus, `plotMetrics` allows us to go one step further and examine biological differences between individual groups of cells. 

Here,
the `group` parameter defines a grouping variable, present as a column of metadata in the single cell object. 
the `factor` parameter defines a continuous variable to plot per level of the `group` parameter. 
the `threshold` parameter defines the minimum number of cells required for each level of the `group` parameter. 

### plotMetrics - clusters
```{r}
plotMetrics(test.sce, group = "seurat_clusters", factor = "nCount_RNA", threshold = 10)
```
Using the `trans` parameter, the x axis in `plotMetrics` can be transformed using standard methods available within `R`

```{r}
plotMetrics(test.sce, group = "seurat_clusters", factor = "nCount_RNA", threshold = 10, trans = "log10")
```

### plotMetrics - barcodes
`plotMetrics` accepts any grouping variable available in the sample metadata. Here we examine transcriptional differences between clones using lineage barcode information. 

NB: In this test dataset the number of cells per clone is small.

```{r}
plotMetrics(test.sce, group = "barcode", factor = "nCount_RNA", threshold = 2)
```

## 2. Examine number of cells per grouping variable - plotCellsPerGroup

We may also be interested in basic metrics like how many cells comprise a grouping variable of interest such as cluster, celltype or lineage barcode. The `plotCellsPerGroup` function allows this to be easily plotted. 

Groups above a user defined threshold are highlighted.

Here, 
the `group` parameter defines a grouping variable, present as a column of metadata in the single cell object. 
the `threshold` parameter defines the minimum number of cells. Levels of the `group` parameter above this threshold will be labelled.

### plot cells per cluster
```{r}
plotCellsPerGroup(test.sce, group = "seurat_clusters", threshold = 5)
```

### plot cells per lineage barcode
```{r}
plotCellsPerGroup(test.sce, group = "barcode", threshold = 3)
```

## 3. Plot distribution of cells across clusters - plotCellsInClusters
We may also be interested in the distribution of cells within certain groups across levels of another group. For example we may want to reveal the proportion of cells in a certain cell cycle phase across louvain clusters within a single cell dataset. The `plotCellsInClusters` function allows users to examine these questions. 

Here,
the `group` parameter defines a grouping variable, present as a column of metadata in the single cell object. 
the `factor` parameter defines a level of `group` to calculate percentage abundance within each level of `idents`. 
the `idents` parameter defines a second grouping variable, present as a column of metadata in the single cell object.


### Cell cycle phase across clusters
```{r}
plotCellsInClusters(test.sce, group = "Phase", factor = "G1", clusters = "seurat_clusters")
```

The `plotCellsInClusters` framework can extend to any discrete variables present in the dataset. Here we examine the representation of lineage barcode BC_12904 across clusters

```{r}
plotCellsInClusters(test.sce, group = "barcode", factor = "BC_12904", clusters = "seurat_clusters")
```

## 4. Determine enrichment within clusters - plotClusterEnrichment
To check whether a clone is enriched in a cluster or cell type, we can perform a hypergeometric test using the `plotClusterEnrichment` function. Here we test for enrichment of cells in G2M cell cycle phase across louvain clusters.

Here,
the `group` parameter defines a grouping variable, present as a column of metadata in the single cell object. 
the `factor` parameter defines a level of the `group` parameter to test for enrichment within each level of `idents`. 
the `idents` parameter defines a second grouping variable to test for enrichment of `factor` at each level.
the `threshold` parameter defines a p-value threshold for the hypergeometric test.

```{r}
plotClusterEnrichment(
  test.sce,
  group = "Phase",
  factor = "G2M",
  clusters = "seurat_clusters",
  threshold = 0.01,
  order = T,
  plot = T
)
```

## 5. Create pseudo-bulk dataset for differential expression analysis - pseudoBulk
We can also use the gene expression values for downstream analyses. When the cells come from samples with different conditions (e.g. Treatment vs. Control), one approach is to use the `pseudoBulk` function to aggregate cell counts from each sample into pseudo-bulk counts and perform differential expression analysis.

To demonstrate, first, we can assign our cells in the example single-cell dataset to different samples of origin. In this example, we create 6 samples and each of the samples has a group assignment ("Treatment" or "Control").

```{r}
# Create 6 samples and assign them to Treatment or Control groups.
sample_ids <- c(paste("Treatment", 1:3, sep = "_"), paste("Control", 1:3, sep = "_"))
group <- c(rep("Treatment", 3), rep("Control", 3))

# Assign cells to samples and the corresponding groups.
set.seed(1)
cell_assignments <- sample(sample_ids, ncol(test.sce), replace = TRUE)
group_map <- setNames(group, sample_ids)
assigned_groups <- group_map[cell_assignments]
colData(test.sce)[,c("Sample_id", "Group")] <- DataFrame(cell_assignments, assigned_groups)
```

Now, we can create the pseudo-bulk counts using the `pseudoBulk` function. Here,
the `sample` parameter defines the samples of origin of single cells, present as a column of metadata in the single cell object.
the `group` parameter defines the groups that the `sample`s belong to. The function checks that each sample has the same group information for all corresponding cells. 
the `barcode` parameter takes a vector of two elements where the first element is the name of the barcode column from the metadata, and the second element is the specific barcode ID to filter cells by. When this argument is supplied, the function calculates pseudo-bulk counts only for the cells with the specified barcode.
the `threshold` parameter defines the minimum acceptable number of samples for each group to have. It is only used if the group argument is not `NULL`. It requires a value of at least 2.

```{r}
source("/Users/salihaey/Desktop/bartools/pseudoBulk.R")
dge<- pseudoBulk(test.sce,
                 sample  = "Sample_id",
                 group = "Group",
                 barcode = NULL,
                 threshold = 3)
dge
```

As we provided information about sample groups, we can perform differential expression analysis with the resulting DGEList object using `edgeR`.

```{r}
dge <- calcNormFactors(dge)
design <- model.matrix(~ group, data=dge$samples)
dge <- estimateDisp(dge, design)
fit <- glmQLFit(dge, design)
res <- glmQLFTest(fit, coef=2)
topTags(res)
```

We can also create a pseudo-bulk object for a particular barcode.

```{r}
dge<- pseudoBulk(test.sce, 
                 sample  = "Sample_id",
                 barcode = c("barcode", "BC_18120"))
dge
```

## 6. Session Info

```{r}
sessionInfo()
```
