<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>bartools: scRNA-seq analysis guide • bartools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="bartools: scRNA-seq analysis guide">
<meta property="og:description" content="bartools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bartools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/bartools_quickstart.html">bartools: quickstart guide</a>
    </li>
    <li>
      <a href="../articles/bartools_single_cell_analysis.html">bartools: scRNA-seq analysis guide</a>
    </li>
    <li>
      <a href="../articles/import_qc_single_cell.html">bartools: import and QC single-cell barcode data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>bartools: scRNA-seq analysis guide</h1>
                        <h4 data-toc-skip class="author">Henrietta
Holze</h4>
            
            <h4 data-toc-skip class="date">September 05, 2023</h4>
      
      
      <div class="hidden name"><code>bartools_single_cell_analysis.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="bartools-single-cell-guide">Bartools single-cell guide<a class="anchor" aria-label="anchor" href="#bartools-single-cell-guide"></a>
</h2>
<p>The <code>bartools</code> package contains methods to simplify clone
level analyses from single-cell cellular barcoding datasets. The purpose
of this vignette is to highlight these capabilities of
<code>bartools</code>.</p>
<div class="section level3">
<h3 id="setup">0. Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h3>
<p>This vignette makes use of a simple test single cell experiment
object (SCE) included in the <code>bartools</code> package. The dataset
contains 100 cells each with 100 randomly sampled genes. Each cell has
lineage barcode information annotated in the
<code>test.sce$barcode</code> field.</p>
<p>Dimensionality reduction and clustering were previously performed
using <code>Seurat</code>. Cluster assignments are in the
<code>test.sce$seurat_clusters</code> field.</p>
<div class="section level4">
<h4 id="load-the-bartools-package">Load the <code>bartools</code> package<a class="anchor" aria-label="anchor" href="#load-the-bartools-package"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">bartools</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: edgeR</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'edgeR' was built under R version 4.2.2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: limma</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'limma' was built under R version 4.2.2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ggplot2</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/knitr/man/opts_chunk.html" class="external-link">opts_chunk</a></span><span class="op">$</span><span class="fu">set</span><span class="op">(</span>dev<span class="op">=</span><span class="st">"png"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="load-the-test-sce-dataset">Load the <code>test.sce</code> dataset<a class="anchor" aria-label="anchor" href="#load-the-test-sce-dataset"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">test.sce</span><span class="op">)</span></span>
<span><span class="va">test.sce</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: SingleCellExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'SingleCellExperiment' was built under R version 4.2.2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'SummarizedExperiment' was built under R version 4.2.1</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'MatrixGenerics' was built under R version 4.2.1</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'GenomicRanges' was built under R version 4.2.2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'BiocGenerics' was built under R version 4.2.1</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:limma':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     plotMA</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'S4Vectors' was built under R version 4.2.2</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'IRanges' was built under R version 4.2.1</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'GenomeInfoDb' was built under R version 4.2.2</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package 'Biobase' was built under R version 4.2.1</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'SingleCellExperiment'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:edgeR':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     cpm</span></span></code></pre>
<pre><code><span><span class="co">## class: SingleCellExperiment </span></span>
<span><span class="co">## dim: 100 100 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(3): counts logcounts scaledata</span></span>
<span><span class="co">## rownames(100): Mrpl15 Lypla1 ... Tmem131 Cnga3</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(100): GCTACCTAGAGGCCAT ACTGCAAGTGATTCAC ... CGAGTTACAGCGACAA</span></span>
<span><span class="co">##   TTTATGCGTACAGTTC</span></span>
<span><span class="co">## colData names(27): orig.ident nCount_RNA ... seurat_clusters ident</span></span>
<span><span class="co">## reducedDimNames(3): PCA TSNE UMAP</span></span>
<span><span class="co">## mainExpName: RNA</span></span>
<span><span class="co">## altExpNames(1): HTO</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="analyse-dataset-metrics-using-plotmetrics">1. Analyse dataset metrics using plotMetrics<a class="anchor" aria-label="anchor" href="#analyse-dataset-metrics-using-plotmetrics"></a>
</h3>
<p>Single cell RNA sequencing datasets can reveal transcriptional
differences between groups of cells / celltypes that are of biological
importance. The <code>plotMetrics</code> function takes a single cell
object in Seurat or SingleCellExperiment format and plots a desired
continuous variable (e.g. number of transcripts or genes detected per
cell) split across any groups of interest. These groups could be
clusters, celltypes or, with lineage barcode information, individual
clones. Thus, <code>plotMetrics</code> allows us to go one step further
and examine biological differences between individual groups of
cells.</p>
<p>Here, the <code>group</code> parameter defines a grouping variable,
present as a column of metadata in the single cell object. the
<code>factor</code> parameter defines a continuous variable to plot per
level of the <code>group</code> parameter. the <code>threshold</code>
parameter defines the minimum number of cells required for each level of
the <code>group</code> parameter.</p>
<div class="section level4">
<h4 id="plotmetrics---clusters">plotMetrics - clusters<a class="anchor" aria-label="anchor" href="#plotmetrics---clusters"></a>
</h4>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotMetrics.html">plotMetrics</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"seurat_clusters"</span>, factor <span class="op">=</span> <span class="st">"nCount_RNA"</span>, threshold <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-3-1.png" width="700">
Using the <code>trans</code> parameter, the x axis in
<code>plotMetrics</code> can be transformed using standard methods
available within <code>R</code></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotMetrics.html">plotMetrics</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"seurat_clusters"</span>, factor <span class="op">=</span> <span class="st">"nCount_RNA"</span>, threshold <span class="op">=</span> <span class="fl">10</span>, trans <span class="op">=</span> <span class="st">"log10"</span><span class="op">)</span></span></code></pre></div>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="plotmetrics---barcodes">plotMetrics - barcodes<a class="anchor" aria-label="anchor" href="#plotmetrics---barcodes"></a>
</h4>
<p><code>plotMetrics</code> accepts any grouping variable available in
the sample metadata. Here we examine transcriptional differences between
clones using lineage barcode information.</p>
<p>NB: In this test dataset the number of cells per clone is small.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotMetrics.html">plotMetrics</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"barcode"</span>, factor <span class="op">=</span> <span class="st">"nCount_RNA"</span>, threshold <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="examine-number-of-cells-per-grouping-variable---plotcellspergroup">2. Examine number of cells per grouping variable -
plotCellsPerGroup<a class="anchor" aria-label="anchor" href="#examine-number-of-cells-per-grouping-variable---plotcellspergroup"></a>
</h3>
<p>We may also be interested in basic metrics like how many cells
comprise a grouping variable of interest such as cluster, celltype or
lineage barcode. The <code>plotCellsPerGroup</code> function allows this
to be easily plotted.</p>
<p>Groups above a user defined threshold are highlighted.</p>
<p>Here, the <code>group</code> parameter defines a grouping variable,
present as a column of metadata in the single cell object. the
<code>threshold</code> parameter defines the minimum number of cells.
Levels of the <code>group</code> parameter above this threshold will be
labelled.</p>
<div class="section level4">
<h4 id="plot-cells-per-cluster">plot cells per cluster<a class="anchor" aria-label="anchor" href="#plot-cells-per-cluster"></a>
</h4>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellsPerGroup.html">plotCellsPerGroup</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"seurat_clusters"</span>, threshold <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="plot-cells-per-lineage-barcode">plot cells per lineage barcode<a class="anchor" aria-label="anchor" href="#plot-cells-per-lineage-barcode"></a>
</h4>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellsPerGroup.html">plotCellsPerGroup</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"barcode"</span>, threshold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="plot-distribution-of-cells-across-clusters---plotcellsinclusters">3. Plot distribution of cells across clusters -
plotCellsInClusters<a class="anchor" aria-label="anchor" href="#plot-distribution-of-cells-across-clusters---plotcellsinclusters"></a>
</h3>
<p>We may also be interested in the distribution of cells within certain
groups across levels of another group. For example we may want to reveal
the proportion of cells in a certain cell cycle phase across louvain
clusters within a single cell dataset. The
<code>plotCellsInClusters</code> function allows users to examine these
questions.</p>
<p>Here, the <code>group</code> parameter defines a grouping variable,
present as a column of metadata in the single cell object. the
<code>factor</code> parameter defines a level of <code>group</code> to
calculate percentage abundance within each level of <code>idents</code>.
the <code>idents</code> parameter defines a second grouping variable,
present as a column of metadata in the single cell object.</p>
<div class="section level4">
<h4 id="cell-cycle-phase-across-clusters">Cell cycle phase across clusters<a class="anchor" aria-label="anchor" href="#cell-cycle-phase-across-clusters"></a>
</h4>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellsInClusters.html">plotCellsInClusters</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"Phase"</span>, factor <span class="op">=</span> <span class="st">"G1"</span>, clusters <span class="op">=</span> <span class="st">"seurat_clusters"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 7 × 2</span></span></span>
<span><span class="co">##   seurat_clusters     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 0                  17</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 1                  15</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> 2                  15</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> 3                   6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> 4                   4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> 5                   3</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> 7                   1</span></span></code></pre>
<pre><code><span><span class="co">## Warning in ggplot2::geom_histogram(stat = "identity"): Ignoring unknown</span></span>
<span><span class="co">## parameters: `binwidth`, `bins`, and `pad`</span></span></code></pre>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>The <code>plotCellsInClusters</code> framework can extend to any
discrete variables present in the dataset. Here we examine the
representation of lineage barcode BC_12904 across clusters</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCellsInClusters.html">plotCellsInClusters</a></span><span class="op">(</span><span class="va">test.sce</span>, group <span class="op">=</span> <span class="st">"barcode"</span>, factor <span class="op">=</span> <span class="st">"BC_12904"</span>, clusters <span class="op">=</span> <span class="st">"seurat_clusters"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">##   seurat_clusters     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 0                   2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 1                   1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> 7                   1</span></span></code></pre>
<pre><code><span><span class="co">## Warning in ggplot2::geom_histogram(stat = "identity"): Ignoring unknown</span></span>
<span><span class="co">## parameters: `binwidth`, `bins`, and `pad`</span></span></code></pre>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="determine-enrichment-within-clusters---plotclusterenrichment">4. Determine enrichment within clusters - plotClusterEnrichment<a class="anchor" aria-label="anchor" href="#determine-enrichment-within-clusters---plotclusterenrichment"></a>
</h3>
<p>To check whether a clone is enriched in a cluster or cell type, we
can perform a hypergeometric test using the
<code>plotClusterEnrichment</code> function. Here we test for enrichment
of cells in G2M cell cycle phase across louvain clusters.</p>
<p>Here, the <code>group</code> parameter defines a grouping variable,
present as a column of metadata in the single cell object. the
<code>factor</code> parameter defines a level of the <code>group</code>
parameter to test for enrichment within each level of
<code>idents</code>. the <code>idents</code> parameter defines a second
grouping variable to test for enrichment of <code>factor</code> at each
level. the <code>threshold</code> parameter defines a p-value threshold
for the hypergeometric test.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotClusterEnrichment.html">plotClusterEnrichment</a></span><span class="op">(</span></span>
<span>  <span class="va">test.sce</span>,</span>
<span>  group <span class="op">=</span> <span class="st">"Phase"</span>,</span>
<span>  factor <span class="op">=</span> <span class="st">"G2M"</span>,</span>
<span>  clusters <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  threshold <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  order <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  plot <span class="op">=</span> <span class="cn">T</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The following ident levels had no observations and were removed: 6</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_0</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 24</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 0</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.424242424242424"</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_1</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 27</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 0</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.469090909090909"</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_2</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 28</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 1</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.0763636363636364"</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_3</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 10</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 1</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.0090909090909091"</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_4</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 5</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 0</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.0979797979797982"</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_5</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 5</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 0</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.0979797979797982"</span></span></code></pre>
<pre><code><span><span class="co">## ---</span></span></code></pre>
<pre><code><span><span class="co">## cluster_7</span></span></code></pre>
<pre><code><span><span class="co">## all cells: 100</span></span></code></pre>
<pre><code><span><span class="co">## all G2M cells: 2</span></span></code></pre>
<pre><code><span><span class="co">## universe: 98</span></span></code></pre>
<pre><code><span><span class="co">## cluster total cells: 1</span></span></code></pre>
<pre><code><span><span class="co">## cluster G2M cells: 0</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Hypergeometric test p-value: 0.0200000000000001"</span></span></code></pre>
<p><img src="bartools_single_cell_analysis_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="create-pseudo-bulk-dataset-for-differential-expression-analysis---pseudobulk">5. Create pseudo-bulk dataset for differential expression analysis -
pseudoBulk<a class="anchor" aria-label="anchor" href="#create-pseudo-bulk-dataset-for-differential-expression-analysis---pseudobulk"></a>
</h3>
<p>We can also use the gene expression values for downstream analyses.
When the cells come from samples with different conditions
(e.g. Treatment vs. Control), one approach is to use the
<code>pseudoBulk</code> function to aggregate cell counts from each
sample into pseudo-bulk counts and perform differential expression
analysis.</p>
<p>To demonstrate, first, we can assign our cells in the example
single-cell dataset to different samples of origin. In this example, we
create 6 samples and each of the samples has a group assignment
(“Treatment” or “Control”).</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create 6 samples and assign them to Treatment or Control groups.</span></span>
<span><span class="va">sample_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Assign cells to samples and the corresponding groups.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">cell_assignments</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">sample_ids</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">test.sce</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">group_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">group</span>, <span class="va">sample_ids</span><span class="op">)</span></span>
<span><span class="va">assigned_groups</span> <span class="op">&lt;-</span> <span class="va">group_map</span><span class="op">[</span><span class="va">cell_assignments</span><span class="op">]</span></span>
<span><span class="fu">colData</span><span class="op">(</span><span class="va">test.sce</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample_id"</span>, <span class="st">"Group"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">DataFrame</span><span class="op">(</span><span class="va">cell_assignments</span>, <span class="va">assigned_groups</span><span class="op">)</span></span></code></pre></div>
<p>Now, we can create the pseudo-bulk counts using the
<code>pseudoBulk</code> function. Here, the <code>sample</code>
parameter defines the samples of origin of single cells, present as a
column of metadata in the single cell object. the <code>group</code>
parameter defines the groups that the <code>sample</code>s belong to.
The function checks that each sample has the same group information for
all corresponding cells. the <code>barcode</code> parameter takes a
vector of two elements where the first element is the name of the
barcode column from the metadata, and the second element is the specific
barcode ID to filter cells by. When this argument is supplied, the
function calculates pseudo-bulk counts only for the cells with the
specified barcode. the <code>threshold</code> parameter defines the
minimum acceptable number of samples for each group to have. It is only
used if the group argument is not <code>NULL</code>. It requires a value
of at least 2.</p>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/source.html" class="external-link">source</a></span><span class="op">(</span><span class="st">"/Users/salihaey/Desktop/bartools/pseudoBulk.R"</span><span class="op">)</span></span>
<span><span class="va">dge</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudoBulk.html">pseudoBulk</a></span><span class="op">(</span><span class="va">test.sce</span>,</span>
<span>                 sample  <span class="op">=</span> <span class="st">"Sample_id"</span>,</span>
<span>                 group <span class="op">=</span> <span class="st">"Group"</span>,</span>
<span>                 barcode <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 threshold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">dge</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class "DGEList"</span></span>
<span><span class="co">## $counts</span></span>
<span><span class="co">##         Control_1 Control_2 Control_3 Treatment_1 Treatment_2 Treatment_3</span></span>
<span><span class="co">## Mrpl15         39        43        60          59          54          38</span></span>
<span><span class="co">## Lypla1          8        14        15           8          16          11</span></span>
<span><span class="co">## Gm37988         0         0         0           0           0           0</span></span>
<span><span class="co">## Tcea1          11        18        25          14          18          16</span></span>
<span><span class="co">## Atp6v1h        14         6         7           6          10           2</span></span>
<span><span class="co">## 95 more rows ...</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $samples</span></span>
<span><span class="co">##        sample     group</span></span>
<span><span class="co">## 1   Control_1   Control</span></span>
<span><span class="co">## 2   Control_2   Control</span></span>
<span><span class="co">## 3   Control_3   Control</span></span>
<span><span class="co">## 4 Treatment_1 Treatment</span></span>
<span><span class="co">## 5 Treatment_2 Treatment</span></span>
<span><span class="co">## 6 Treatment_3 Treatment</span></span></code></pre>
<p>As we provided information about sample groups, we can perform
differential expression analysis with the resulting DGEList object using
<code>edgeR</code>.</p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">group</span>, data<span class="op">=</span><span class="va">dge</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateDisp.html" class="external-link">estimateDisp</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html" class="external-link">glmQLFit</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html" class="external-link">glmQLFTest</a></span><span class="op">(</span><span class="va">fit</span>, coef<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html" class="external-link">topTags</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Coefficient:  groupTreatment </span></span>
<span><span class="co">##                    logFC   logCPM        F     PValue FDR</span></span>
<span><span class="co">## Terf1         -0.8459600 12.83750 5.003634 0.02584451   1</span></span>
<span><span class="co">## Rpl7          -0.1593976 18.82013 3.154959 0.07645669   1</span></span>
<span><span class="co">## Rb1cc1         0.6810149 12.59832 2.618375 0.10642078   1</span></span>
<span><span class="co">## Mcm3           0.3579472 14.08153 2.592255 0.10817621   1</span></span>
<span><span class="co">## 4732440D04Rik -1.7321077 10.80961 2.445832 0.11862814   1</span></span>
<span><span class="co">## Cox5b         -0.1573944 17.05650 2.111197 0.14700894   1</span></span>
<span><span class="co">## Rrs1           0.4296893 13.19945 1.891008 0.16985800   1</span></span>
<span><span class="co">## Ube2w          0.4112542 13.23263 1.789296 0.18177197   1</span></span>
<span><span class="co">## Mybl1          2.6930194 10.15808 1.680259 0.19563746   1</span></span>
<span><span class="co">## Kcnq5          2.6921936 10.15801 1.679529 0.19573432   1</span></span></code></pre>
<p>We can also create a pseudo-bulk object for a particular barcode.</p>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dge</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/pseudoBulk.html">pseudoBulk</a></span><span class="op">(</span><span class="va">test.sce</span>, </span>
<span>                 sample  <span class="op">=</span> <span class="st">"Sample_id"</span>,</span>
<span>                 barcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"barcode"</span>, <span class="st">"BC_18120"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Removing samples with less than 2 cells."</span></span></code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dge</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class "DGEList"</span></span>
<span><span class="co">## $counts</span></span>
<span><span class="co">##         Control_1 Control_3</span></span>
<span><span class="co">## Mrpl15         12         8</span></span>
<span><span class="co">## Lypla1          2         0</span></span>
<span><span class="co">## Gm37988         0         0</span></span>
<span><span class="co">## Tcea1           1         5</span></span>
<span><span class="co">## Atp6v1h         3         1</span></span>
<span><span class="co">## 95 more rows ...</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $samples</span></span>
<span><span class="co">##      sample</span></span>
<span><span class="co">## 1 Control_1</span></span>
<span><span class="co">## 2 Control_3</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="session-info">6. Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SingleCellExperiment_1.20.1 SummarizedExperiment_1.28.0</span></span>
<span><span class="co">##  [3] Biobase_2.58.0              GenomicRanges_1.50.2       </span></span>
<span><span class="co">##  [5] GenomeInfoDb_1.34.9         IRanges_2.32.0             </span></span>
<span><span class="co">##  [7] S4Vectors_0.36.2            BiocGenerics_0.44.0        </span></span>
<span><span class="co">##  [9] MatrixGenerics_1.10.0       matrixStats_1.0.0          </span></span>
<span><span class="co">## [11] bartools_1.0.0              ggplot2_3.4.4              </span></span>
<span><span class="co">## [13] edgeR_3.40.2                limma_3.54.2               </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] viridis_0.6.4          sass_0.4.7             viridisLite_0.4.2     </span></span>
<span><span class="co">##  [4] jsonlite_1.8.7         splines_4.2.0          bslib_0.5.1           </span></span>
<span><span class="co">##  [7] highr_0.10             GenomeInfoDbData_1.2.9 ggrepel_0.9.4         </span></span>
<span><span class="co">## [10] yaml_2.3.7             pillar_1.9.0           lattice_0.22-5        </span></span>
<span><span class="co">## [13] glue_1.6.2             digest_0.6.33          XVector_0.38.0        </span></span>
<span><span class="co">## [16] colorspace_2.1-0       htmltools_0.5.6.1      Matrix_1.6-4          </span></span>
<span><span class="co">## [19] pkgconfig_2.0.3        zlibbioc_1.44.0        purrr_1.0.2           </span></span>
<span><span class="co">## [22] scales_1.2.1           tibble_3.2.1           mgcv_1.9-0            </span></span>
<span><span class="co">## [25] farver_2.1.1           generics_0.1.3         cachem_1.0.8          </span></span>
<span><span class="co">## [28] withr_2.5.2            cli_3.6.1              magrittr_2.0.3        </span></span>
<span><span class="co">## [31] memoise_2.0.1          evaluate_0.23          fs_1.6.3              </span></span>
<span><span class="co">## [34] fansi_1.0.5            nlme_3.1-163           MASS_7.3-60           </span></span>
<span><span class="co">## [37] vegan_2.6-4            textshaping_0.3.7      tools_4.2.0           </span></span>
<span><span class="co">## [40] ineq_0.2-13            lifecycle_1.0.3        stringr_1.5.0         </span></span>
<span><span class="co">## [43] munsell_0.5.0          locfit_1.5-9.8         DelayedArray_0.24.0   </span></span>
<span><span class="co">## [46] cluster_2.1.4          compiler_4.2.0         pkgdown_2.0.7         </span></span>
<span><span class="co">## [49] jquerylib_0.1.4        systemfonts_1.0.5      rlang_1.1.1           </span></span>
<span><span class="co">## [52] grid_4.2.0             RCurl_1.98-1.12        rstudioapi_0.15.0     </span></span>
<span><span class="co">## [55] labeling_0.4.3         bitops_1.0-7           rmarkdown_2.25        </span></span>
<span><span class="co">## [58] gtable_0.3.4           R6_2.5.1               gridExtra_2.3         </span></span>
<span><span class="co">## [61] knitr_1.45             dplyr_1.1.3            fastmap_1.1.1         </span></span>
<span><span class="co">## [64] utf8_1.2.4             rprojroot_2.0.3        ragg_1.2.6            </span></span>
<span><span class="co">## [67] permute_0.9-7          desc_1.4.2             stringi_1.7.12        </span></span>
<span><span class="co">## [70] parallel_4.2.0         Rcpp_1.0.11            vctrs_0.6.4           </span></span>
<span><span class="co">## [73] tidyselect_1.2.0       xfun_0.41</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Dane Vassiliadis, Henrietta Holze.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
